Prox/Prox/Models/Place.swift

```diff
@@ -388,58 +388,79 @@ struct OpenHours {
         }
+        return allOpeningTimes.sorted { (periodA, periodB) in
+            return  periodA.openTime < periodB.openTime
+        }
+    }
+
+    private func getCurrentOpeningPeriod(fromOpeningPeriods openingPeriods: [OpenPeriod], forDate date: Date) -> OpenPeriod? {
+        let currentPeriods = openingPeriods.filter { isTime(date, duringOpeningPeriod: $0) }
+        return currentPeriods.first
+    }
+
+    private func getNextOpeningPeriod(fromOpeningPeriods openingPeriods: [OpenPeriod], forDate date: Date) -> OpenPeriod? {
+        var nextOpenPeriod: OpenPeriod? = nil
+        for openPeriod in openingPeriods {
+            if openPeriod.openTime > date {
+                guard let currentOpenTime = nextOpenPeriod else {
+                    nextOpenPeriod = openPeriod
+                    break
+                }
 
-        // Add 24 hours to the closing time if the closing time if before the opening time
-        if closingTime <= openingTime {
-            closingTime += AppConstants.ONE_DAY
+                if openPeriod.openTime < currentOpenTime.openTime {
+                    nextOpenPeriod = openPeriod
+                }
+            }
         }
-        return  closingTime
+
+        return nextOpenPeriod
+    }
+
+    private func getEarliestOpeningPeriod(fromOpeningPeriods periods: [OpenPeriod]) -> OpenPeriod? {
+        return periods.first
     }
 
-    /** 
-    * The logic for isOpen is as follows:
-    * Apply opening time and closing time to the current date (provided as an argument)
-    * If the closing time is < the opening time, then the venue is open over midnight and we have to add 24hrs to the closing time
-    * If the current time is before the opening time, but the venue closed before midnight the previous day, the venue is closed
-    * If the current time if before the opening time, but the venue closed after midnight the previous day, the venue is closed if the 
-      current time is after yesterdays closing time, otherwise the venue is open
-    * If the current time is before the closing time and after the opening time, the venue is open
-    * If the current time is after the closing time and before the opening time, the venue is closed
-    **/
+
     func isOpen(atTime time: Date) -> Bool {
-        guard let openingTime = getOpeningTime(forDate: time),
-            let closingTime = getClosingTime(forDate: time) else {
+        guard let allOpeningTimes = getOpeningTimes(forDate: time) else {
             return false
         }
 
-        // Is the current time before the opening time and the venue closed after midnight the previous day?
-        if time < openingTime,
-            let yesterdaysClosingTime = getClosingTime(forDate: time.addingTimeInterval(-AppConstants.ONE_DAY)),
-            yesterdaysClosingTime < openingTime {
+        guard let _ = getCurrentOpeningPeriod(fromOpeningPeriods: allOpeningTimes, forDate: time) else {
```

Comment:
nit: It's confusing what you're trying to guard against. Perhaps you could clarify with a named variable:
```swift
let isOpen = getCurrentOpeningPeriod... != nil
guard isOpen else {
```

It could also help to return early so I don't have to keep the stored state in my head:
```swift
let isOpen = ...
if isOpen {
    return true
}

// do not open stuff
```